### PR TITLE
[Add]: Relationship between PriorityClass and other k8s components

### PR DESCRIPTION
**Pull request by pandeyshubham03**
**Notes for Reviewers**

This PR fixes #
1. Adds relationship between PriorityClass and other k8s components.

https://github.com/meshery/meshery/assets/128715606/84ae25eb-b32b-4b7c-9cdc-d1afe3a29639

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -1090,6 +1090,40 @@
...
+        "from":[
+          {
+            "kind": "PriorityClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "PriorityClass in Kubernetes defines scheduling priority for Pods."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1105]: Please describe the relationship in addition to the component.

**Comment by RipulHandoo** [comment]
@pandeyshubham03 PriorityClass can indeed have relationships with various Kubernetes components, including Deployments, Pod Template and others.

**Comment by pandeyshubham03** [comment]
I have added the description for the relationships added @leecalcote and I have also added all the components that could have a relationships with PriorityClass @RipulHandoo.

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -1327,6 +1327,118 @@
...
+        "from":[
+          {
+            "kind": "PriorityClass",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["name"]
+              ],
+              "description": "PriorityClass in Kubernetes assigns importance levels to Pods, influencing scheduling decisions. The name of the PriorityClass is referenced by other k8s components"
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1342]: Let's not abbreviate "Kubernetes" or if we do, make sure that we abide by their branding request that "K8s" be capitalized. Having said that, let's not abbreviate Kubernetes. The PriorityClass name is referenced by other components... for what purpose?

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 3]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "PodTemplate",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "spec", "priorityClassName"]
+              ],
+              "description": "PodTemplate specifies the name of priorityClass, guiding Kubernetes controller to assign priority to Pods created from it, influencing scheduling behavior accordingly."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1366]: "Kubernetes controller" - make this plural.

"created from it" - "it" who?

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 4]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "StatefulSet",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "StatefulSets manage stateful applications. PriorityClass can be applied to StatefulSets' Pods to prioritize them during scheduling."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1388]: "them" - to what component are you referring?

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 5]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "DaemonSet",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "DaemonSets ensure all nodes run a copy of a Pod. PriorityClass assigns importance, impacting scheduling decisions for DaemonSet-managed Pods."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1399]: "impacting" - in what way?

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 6]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "Job",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "PriorityClass in Kubernetes influences Job scheduling, ensuring high-priority Jobs execute first, critical for resource allocation and workload management."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1421]: Rephrase - ", critical for resource allocation and workload management."

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 7]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "CronJob",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "jobTemplate", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "CronJobs can use PriorityClass to prioritize scheduled job execution, ensuring critical tasks are handled with precedence in Kubernetes."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1432]: CronJobs can or do use?

"handled with precedence in Kubernetes." or "ensuring CronJobs are scheduled for execution in accordance with the precedence specified in the associated PriorityClass."?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 8]
```diff
@@ -1327,6 +1327,118 @@
...
+          },
+          {
+            "kind": "CronJob",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "jobTemplate", "spec", "template", "spec", "priorityClassName"]
+              ],
+              "description": "CronJobs can use PriorityClass to prioritize the scheduled job execution, ensuring CronJobs are scheduled for execution in accordance with the precedence specified in the associated PriorityClass."
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1432]: Nice!

**Review by leecalcote — APPROVED** [review]

**Pull request merged**